bubble_sort_by.rb

```diff
@@ -0,0 +1,19 @@
+def bubble_sort_by(arr)
+  len = arr.length
+  (0...len).each do |i|
+    (i + 1...len).each do |j|
+      next unless yield arr[i], arr[j]
```

Comment:
This is the point where you should check if the yield is positive, not in the example.
When the yield is positive, it means the left item is greater than the right, and so there should be a swap.

What your code currently does is to swap every time the yield is not false or nil, which will not give the correct output.